src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentFieldDisplayDriver.cs

```diff
@@ -126,11 +133,26 @@ Task<IDisplayResult> IContentFieldDisplayDriver.BuildEditorAsync(ContentPart con
 
             var field = contentPart.GetOrCreate<TField>(partFieldDefinition.Name);
 
-            BuildPrefix(typePartDefinition, partFieldDefinition, context.HtmlFieldPrefix);
+            if (field != null)
+            {
+                BuildPrefix(typePartDefinition, partFieldDefinition, context.HtmlFieldPrefix);
+
+                var fieldEditorContext = new BuildFieldEditorContext(contentPart, typePartDefinition, partFieldDefinition, context);
+
+                _typePartDefinition = typePartDefinition;
```

Comment:
I don't understand what you are doing here

Reply:
Just setting the typePartDefinition and partFieldDefinition values so that they can be used in the Factory method. Much like what BuildDisplayAsync is doing already.